yjit/src/core.rs

```diff
@@ -530,9 +530,9 @@ fn get_or_create_iseq_payload(iseq: IseqPtr) -> &'static mut IseqPayload {
 /// Iterate over all existing ISEQs
 pub fn for_each_iseq<F: FnMut(IseqPtr)>(mut callback: F) {
     unsafe extern "C" fn callback_wrapper(iseq: IseqPtr, data: *mut c_void) {
-        let callback: &mut &mut dyn FnMut(IseqPtr) -> bool = unsafe { std::mem::transmute(data) };
+        let callback: &mut &mut dyn FnMut(IseqPtr) -> bool = std::mem::transmute(&mut *data);
```

Comment:
Before the transmute here was doing double duty of also dealing with the `*mut` to `&mut` conversion. See https://rust-lang.github.io/rust-clippy/master/index.html#transmute_ptr_to_ref